pilot/pkg/model/push_context.go

```diff
@@ -481,36 +495,53 @@ func (ps *PushContext) GetAllSidecarScopes() map[string][]*SidecarScope {
 }
 
 // DestinationRule returns a destination rule for a service name in a given domain.
-func (ps *PushContext) DestinationRule(proxy *Proxy, hostname Hostname) *Config {
+func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *Config {
```

Comment:
I see we're passing some 'fake' Service - do we just need the hostname and namespace ? May be cleaner to just pass both instead of a fake service, will confuse people.